tests/serverpod_test_server/test/decode_with_type_test.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+void main() {
+  var protocol = Protocol();
+
+  group('Given a JSON string', () {
+    test(
```

Comment:
Discussion: Would it make more sense to create these tests as serialization roundtrips? Starting from a value that is encoded and then decoded and then ensuring that the value is preserved?